### PR TITLE
fix hack/update-estimator-protobuf.sh failed with Golang v1.18

### LINKED ISSUE #1528: `hack/update-estimator-protobuf.sh` failed with Golang v1.18
**What happened**:
I want to make sure everything ok with Go1.18, and I found that the `hack/update-estimator-protobuf.sh` can not work as usual.

```bash
-bash-4.2# hack/update-estimator-protobuf.sh 
topological order bytes
<...>
topological order github.com/karmada-io/karmada/pkg/estimator/pb
2022/03/24 09:34:53 github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/generated.proto:10:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
2022/03/24 09:34:55 github.com/karmada-io/karmada/vendor/k8s.io/api/core/v1/generated.proto:12:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
24 Mar 22 09:34 CST INF Starting mockery dry-run=false version=v2.9.4
24 Mar 22 09:34 CST INF Walking dry-run=false version=v2.9.4
2022/03/24 09:34:59 internal error: package "context" without types was imported from "github.com/karmada-io/karmada/pkg/estimator/service"
pkg/estimator/service/doc.go:4: running "mockery": exit status 1
```
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
Just run `hack/update-estimator-protobuf.sh` with Go 1.18.

**Anything else we need to know?**:

**Environment**:
- Karmada version: master (https://github.com/karmada-io/karmada/commit/f8620a9fee999ed24f977f3d5523f8864445de8f)
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`): N/A
- Others:

cc @GarryBest

### PR DESCRIPTION
**Pull request by Garrybest**
Signed-off-by: Garrybest <garrybest@foxmail.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Refer to https://github.com/vektra/mockery/issues/434#issuecomment-1076091305.

**Which issue(s) this PR fixes**:
Fixes #1528

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/update-estimator-protobuf.sh [hunk 1]
```diff
@@ -14,7 +14,7 @@ export PATH=$PATH:$GOPATH/bin
 GO111MODULE=on go install golang.org/x/tools/cmd/goimports
 GO111MODULE=on go install k8s.io/code-generator/cmd/go-to-protobuf
 GO111MODULE=on go install github.com/gogo/protobuf/protoc-gen-gogo
-GO111MODULE=on go install github.com/vektra/mockery/v2
+GO111MODULE=on go install github.com/vektra/mockery/v2@v2.10.0
```
Comment (RainbowMango) [inline hack/update-estimator-protobuf.sh:17]: If it is due to the version.
I guess we should update the version at the [go.mod](https://github.com/karmada-io/karmada/blob/86649395b95a08f1b350281ec105a0aa9f7a89f0/go.mod#L17). Not here.

Comment (Garrybest) [reply hack/update-estimator-protobuf.sh:17]: That's right, thanks for correction.

Comment (RainbowMango) [inline hack/update-estimator-protobuf.sh:17]:
```suggestion
GO111MODULE=on go install github.com/vektra/mockery/v2
```
This change should be revert here, `go install` will install the binary according to `go.sum` where we manages the dependencies.

Comment (Garrybest) [reply hack/update-estimator-protobuf.sh]: I forgot to make a change.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**